x/masterchef/migrations/v5_migration.go

```diff
@@ -0,0 +1,14 @@
+package migrations
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (m Migrator) V5Migration(ctx sdk.Context) error {
+	pools := m.keeper.GetAllPoolInfos(ctx)
+	for _, pool := range pools {
+		pool.EnableEdenRewards = true
+		m.keeper.SetPoolInfo(ctx, pool)
+	}
```

Comment:
this migration will fail, we need to have legacy structure when loading and then new structure when setting.
Please update, for reference you can see [this](https://github.com/elys-network/elys/pull/755/files#diff-0aaa985b1adf1dcd60053a3cc5d0ee45c9bd50e685cd44ee869dd4989be492a3)

Reply:
is this updated ?

Reply:
Hello @amityadav0  yes that was uptated. 

x/masterchef/migrations/v5_migration.go

Reply:
Hey @fenriz07, I don't think this migration will pass because loading logic is based on new position but we want to load legacy, have you tested this script ?